### PR TITLE
Task/g73 apply catabuffer

### PR DESCRIPTION
**Pull request by rg911**
### Apply Catbuffer

- merged changes in [PR#71](https://github.com/nemtech/nem2-sdk-java/pull/71) (which can be closed now)
- Generated latest catbuffer from the generator
- Applied catbuffer to this SDK
- Removed flatbuffer staffs (schema, builder etc...)
- Updated models and transaction to cope with new catbuffer
- Applied `Dragon` features.
- Fixed all unit tests
- Added account restriction transactions
- Reset transaction versions to `1` (Elephant)

** Noted that the integration tests might not be fully working, which will be looked at separate PR(s).

Issue #73

### CODE REVIEW

File: src/main/java/io/nem/catapult/builders/AccountAddressRestrictionTransactionBuilder.java [hunk 1]
```diff
@@ -0,0 +1,128 @@
+/**
+*** Copyright (c) 2016-present,
+*** Jaguar0625, gimre, BloodyRookie, Tech Bureau, Corp. All rights reserved.
```
Comment (evias) [inline src/main/java/io/nem/catapult/builders/AccountAddressRestrictionTransactionBuilder.java]: This file is part of the SDKs not of catapult-server.

File: src/main/java/io/nem/catapult/builders/AccountMosaicRestrictionTransactionBuilder.java [hunk 2]
```diff
@@ -0,0 +1,128 @@
+/**
+*** Copyright (c) 2016-present,
+*** Jaguar0625, gimre, BloodyRookie, Tech Bureau, Corp. All rights reserved.
```
Comment (evias) [inline src/main/java/io/nem/catapult/builders/AccountMosaicRestrictionTransactionBuilder.java]: license comment

File: src/main/java/io/nem/catapult/builders/AccountAddressRestrictionTransactionBuilder.java (hunk 1 above)
Comment (fullcircle23) [reply src/main/java/io/nem/catapult/builders/AccountAddressRestrictionTransactionBuilder.java]: Correct

**Comment by fullcircle23** [comment]
There are some conflicts. I'll resolve them and merge.

**Pull request merged**